Simplified/AxisService/NYPLRSACypher.swift

```diff
@@ -0,0 +1,199 @@
+//
+//  Created by Raman Singh on 2021-04-06.
+//  Copyright © 2021 NYPL Labs. All rights reserved.
+//
+
+import Foundation
+
+#if AXIS
+
+protocol NYPLRSACryptographing {
+  var publicKey: String { get }
+  var privateKey: String { get }
+  var modulus: String { get }
+  var exponent: String { get }
+  func encryptText(_ message: String) -> String?
+  func decryptText(_ encryptedString: String) -> String?
+}
+
+struct NYPLRSACypher: NYPLRSACryptographing {
+  
+  let exponent: String
+  let modulus: String
+  let privateKey: String
+  let publicKey: String
+  private let privateKeySec: SecKey
+  private let publicKeySec: SecKey
+  
+  // MARK: - Static Constants
+  
+  // Key generation constants
+  static private let nyplRSAPrivate = "nypl.rsa.private"
+  static private let nyplRSAPublic = "nypl.rsa.public"
+  
+  // Failed Intitialization constants
+  static private let failedInitializationSummary = "AXIS: Failed to create NYPLRSACypher object!"
+  static private let publicSecKeyGenerationFailure = "Failed to generate public SecKey"
+  static private let privateSecKeyGenerationFailure = "Failed to generate private SecKey"
+  static private let resultPublicKeyGenerationFailure = "Failed to generate resultPublicKey"
+  static private let resultPrivateKeyGenerationFailure = "Failed to generate resultPrivateKey"
+  
+  // Failed Encryption/Decryption constants
+  static private let failedEncryptionSummary = "AXIS: Failed to encrypt text with public key"
+  static private let failedDecryptionSummary = "AXIS: Failed to decrypt text with private key"
+  static private let stringToDataConversionFailure = "Failed to convert message to data"
+  static private let nilEncryptedDataFailure = "encryptedData was nil"
+  static private let nilDecryptedDataFailure = "decryptedData was nil"
+  
+  private static var pub_attributes: CFDictionary = {
+    var keyPairAttr = [NSObject: NSObject]()
+    keyPairAttr[kSecAttrKeyType] = kSecAttrKeyTypeRSA
+    keyPairAttr[kSecAttrKeySizeInBits] = 2048 as NSObject
+    keyPairAttr[kSecPublicKeyAttrs] = publicKeyAttr as NSObject
+    keyPairAttr[kSecPrivateKeyAttrs] = privateKeyAttr as NSObject
+    return keyPairAttr as CFDictionary
+  }()
+  
+  private static let privateKeyAttr: [NSObject: NSObject] = [
+    kSecAttrIsPermanent:true as NSObject,
+    kSecAttrApplicationTag:nyplRSAPrivate.data(using: String.Encoding.utf8)! as NSObject,
+    kSecClass: kSecClassKey,
+    kSecReturnData: kCFBooleanTrue]
+  
+  private static let publicKeyAttr: [NSObject: NSObject] = [
+    kSecAttrIsPermanent:true as NSObject,
+    kSecAttrApplicationTag:nyplRSAPublic.data(using: String.Encoding.utf8)! as NSObject,
+    kSecClass: kSecClassKey,
+    kSecReturnData: kCFBooleanTrue]
+  
+  // MARK: - Initialization
+  init?() {
+    var pubKeySec, privKeySec: SecKey?
+    SecKeyGeneratePair(NYPLRSACypher.pub_attributes, &pubKeySec, &privKeySec)
+    
+    guard let pubSec = pubKeySec else {
+      NYPLErrorLogger.logError(
```

Comment:
minor: there's quite a bit of code repetition in all the error logging, perhaps that can be avoided... it seems like the only thing that changes is the reason

Reply:
Great suggestion. Done